lib/http/preprocessors.js

```diff
@@ -128,9 +125,6 @@ const authHandler = ({ Sessions, Users, Auth }, context) => {
 
     // if POST run authentication as usual but we'll have to check CSRF afterwards.
     return maybeSession.then((cxt) => { // we have to use cxt rather than context for the linter
-      // if authentication failed anyway, just do nothing.
-      if ((cxt == null) || cxt.auth.session.isEmpty()) return;
```

Comment:
Where is an "empty" session now handled?

Reply:
Then shouldn't `authBySessionToken()` be returning the session itself, rather than an `Option`?  And L111 would change from:

```diff
-const maybeSession = authBySessionToken(token, () => reject(Problem.user.authenticationFailed()));
+const session = authBySessionToken(token, () => reject(Problem.user.authenticationFailed()));
```